LoRaEngine/LoraKeysManagerFacade/ClearLnsCache.cs

```diff
@@ -0,0 +1,99 @@
+        {
+            this.edgeDeviceGetter = edgeDeviceGetter;
+            this.serviceClient = serviceClient;
+            this.channelPublisher = channelPublisher;
+            this.logger = logger;
+        }
+
+        [FunctionName(nameof(ClearNetworkServerCache))]
+        public async Task<IActionResult> ClearNetworkServerCache([HttpTrigger(AuthorizationLevel.Function, "post")] HttpRequest req, CancellationToken cancellationToken)
+        {
+            if (req is null) throw new ArgumentNullException(nameof(req));
+
+            try
+            {
+                VersionValidator.Validate(req);
+            }
+            catch (IncompatibleVersionException ex)
+            {
+                return new BadRequestObjectResult(ex.Message);
+            }
+
+            await ClearLnsCacheInternalAsync(cancellationToken);
+
+            return new AcceptedResult();
+        }
+
+        internal async Task ClearLnsCacheInternalAsync(CancellationToken cancellationToken)
+        {
+            this.logger.LogInformation("Clearing device cache for all edge and Pub/Sub channel based Network Servers.");
+            var edgeDevices = await this.edgeDeviceGetter.ListEdgeDevicesAsync(cancellationToken);
+            var tasks = edgeDevices.Select(e => InvokeClearViaDirectMethodAsync(e, cancellationToken)).ToArray();
+
+            await PublishClearMessageAsync();
+            await Task.WhenAll(tasks);
+        }
+
+        internal async Task PublishClearMessageAsync()
+        {
+            await this.channelPublisher.PublishAsync(LoraKeysManagerFacadeConstants.ClearCacheMethodName, new LnsRemoteCall(RemoteCallKind.ClearCache, null));
+            this.logger.LogInformation("Cache clear message published on Pub/Sub channel");
+        }
+
+        internal async Task InvokeClearViaDirectMethodAsync(string lnsId, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var res = await this.serviceClient.InvokeDeviceMethodAsync(lnsId,
+                                                                           LoraKeysManagerFacadeConstants.NetworkServerModuleId,
+                                                                           new CloudToDeviceMethod(LoraKeysManagerFacadeConstants.ClearCacheMethodName),
+                                                                           cancellationToken);
+                if (HttpUtilities.IsSuccessStatusCode(res.Status))
+                {
+                    this.logger.LogInformation("Cache cleared for {gatewayID} via direct method", lnsId);
+                }
+
+                this.logger.LogError("Direct method call to {gatewayID} failed with {statusCode}. Response: {response}",
+                                     lnsId,
+                                     res.Status,
+                                     res.GetPayloadAsJson());
+            }
+            catch (DeviceNotFoundException ex)
```

Comment:
Why hide the exception here instead of relying on the top-level error handling/Function error handling? Wouldn't it be better to let make it visible to the user via the status code that this failed?